### ISSUE TITLE
Failed to run exe file created by py2exe

### ISSUE THREAD

**Issue by rosatheagraham** [py2exe/py2exe#123]
Hello,
my script is working with a webview interface, fed by flask behind it. Therefore, it starts like a desktop programme with an html content inside, so to describe.
It runs normal with command:
`python script.py`

However then,
I had success by creating the script.exe file and the dist folder with py2exe:
`python setup.py install`
`python setup.py py2exe`

Finally, when I run script.exe in the dist folder, **nothing shows** except the line below in the newly created script.log file. The line also appears in the console, when I run python script.py, which tells you the ip and port the flask is serving. 

` _* Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)_`

I would be glad to hear , if you have any idea to detect where the exe stops running.

Note: As I click to script.exe to run, if I refresh a browser with the same address 127.0.0.1:5000 , I observe that something shows in the browser and then disappears. This event indicates, that the exe file starts to run somehow, but does not show anything that shows by running python script.py , such as on the desktop a window and content inside. Could this be a problem with webview? Did I have to add something while using the webview with py2exe? Thanks.

**Comment by albertosottile** [comment]
Hi, thanks for reporting this. Do you think you could provide a minimal working example of your app that I could use to reproduce and debug your issue? Otherwise I am afraid it will be really hard to help you.

**Comment by rosatheagraham** [comment]
Hi, I am very grateful for your help.

I have prepared a minimal working example of my app.
`python script.py` works just normal as you may see.

But the exe created by those commands, doesn't work
`python setup.py install`
`python setup.py py2exe`

Here I made a link: 
https://github.com/rosatheagraham/script

Also there is a zip file which contains my project folder:
[script.zip](https://github.com/py2exe/py2exe/files/7933435/script.zip)

Thank you so much.

**Comment by albertosottile** [comment]
Hi, I am afraid the example is quite far from minimal... It requires a lot of packages including [`webview`](https://pypi.org/project/webview/#description) which I was not able to install in my test environment because `pip install webview` returned several build errors.

Could you try to strip down the MWE a little? I would reduce the dependencies as much as possible and understand what is causing the issue (e.g. remove `webview` or `sqlacademy` or `pywin32` and see if Flask alone is able to function from the embedded binary). If the problem is indeed in `webview`, I am afraid I will not be able to provide any meaningful support...

**Comment by rosatheagraham** [comment]
Hi, Sorry for giving you the trouble. I can't describe you how thankful I am for helping out to examine the problem.
Here is what I have done:

**First option**, for running the script with webview by installing the right packages below (in case you want to test..)

As I try to get blood out of stone, by using flask + webview to make a desktop app, I should have noted down all the prerequisites needed for running the script.

I use python 3.9.10 over windows 10
therefore following packages are required:
`pip install flask`
`pip install wheel`
`pip install pythonnet-2.5.2-cp39-cp39-win_amd64.whl` ,(installed by giving the file path maunally) which the various versions of the file, according to your python version, could be found here:https://www.lfd.uci.edu/~gohlke/pythonlibs/#pythonnet
`pip install pywebview`
`pip install pywin32`
`pip install cefpython3`

Finally the script could be run by : 
`python script.py`

This worked for me. With desktop interface and everything.
(I have removed the arbitrary sqlacademy package from the script. https://github.com/rosatheagraham/script renewed!)

As mentioned earlier in the issue, packing this with py2exe results no interface but a log line such as this: * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)

Notice: in C:\Users\user\AppData\Local\Programs\Python\Python39\Lib\site-packages\py2exe\hooks.py delete the 23th. line : clr (this removal optonal if you want to pack with py2exe)

...


**Second option**, for running the script **_without webview_**:

I have _eliminated the webview from the code_ and made flask run standalone, but in this case there is no desktop appearance which was provided by webview before.
Therefore it's running on the browser when I run : python script.py

Here is the script:
https://github.com/rosatheagraham/script-without-webview

In case, I create an exe by py2exe. The exe makes the flask page running  on http://127.0.0.1:5000 with some missing elements in static folder **on the browser of course.**
Having a log something like that:

```
 * Restarting with stat
 * Debugger is active!
 * Debugger PIN: 186-915-349
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
127.0.0.1 - - [09/Feb/2022 21:23:12] "[32mGET / HTTP/1.1[0m" 302 -
127.0.0.1 - - [09/Feb/2022 21:23:12] "GET /dashboard HTTP/1.1" 200 -
127.0.0.1 - - [09/Feb/2022 21:23:12] "[33mGET /static/vendor/fontawesome-free/css/all.min.css HTTP/1.1[0m" 404 -
127.0.0.1 - - [09/Feb/2022 21:23:12] "[33mGET /css/sb-admin-2.min.css HTTP/1.1[0m" 404 -
```

Which is same with what I get on running python script.py

...

In summary: Indeed, I am looking for an option to run this project with webview. Because the reason I am packing it to exe, _is to have a desktop app_ rather then on browser running.

So here is what I can report for now. Any findings would be great. Thank you very much.

**Comment by albertosottile** [comment]
I apologize, but I do not think I have the bandwidth to help you. The resulting example is still far from minimal (18 MB of various templates, fonts, javascript code...) and it is clear that the issue is only with `pywebview` which is not so popular to justify the debugging investment needed to address this (I was not even able to reproduce it, yet).

The developers of pywebview [recommend to use PyInstaller](https://pywebview.flowrl.com/guide/freezing.html) to freeze their code, hence I second their recommendation and invite you to use PyInstaller instead of py2exe.

Thanks again for having reported this, and good luck with your project.

**Comment by albertosottile** [comment]
Hi, I have somewhat good news. While I was still unable to run your example, I was able to run pywebview examples (https://pywebview.flowrl.com/guide/usage.html) by adding a hook for the `webview` package. You can find wheels from Actions that include this fix here: https://github.com/py2exe/py2exe/actions/runs/1838204537. It would be great if you could download the wheels for your platform, install them, and try your example/script again to see if this fix worked for you (remember to unzip the wheels before running `pip install` on them).

**Comment by rosatheagraham** [comment]
Hi, Sorry it has been a long time to answer your update. I thank you very much again.
Finally I was able to uninstall py2exe and install the new version which you have indicated the wheel with pip, this one for my particular python: py2exe-0.11.1.1-cp39-cp39-win_amd64.whl 

Again success by running `python scripy.py` shows my the webview frame, before packing
but not by the exe created by py2exe

Unfortunately, no success on initiating the webview frame, with the exe produced by py2exe,  **nothing shows** after running the exe

Log file is created with a line
_* Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)_ , which tells me the flask is running inside.

:(

Not giving up! I have done another test by changing the method py2exe setup.py file!

Replacing
`setup(options=options,data_files=data_files,windows=['script.py'])`
With
`setup(options=options,data_files=data_files,console=['script.py'])`

To see if it does give anything more, if I pack the script to start at the console first!

The result is: 

 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
[0401/155724.694:ERROR:icu_util.cc(133)] Invalid file descriptor to ICU data received.
And the console closes immediately.

Dear Alberto, I have no idea if : `[0401/155724.694:ERROR:icu_util.cc(133)] Invalid file descriptor to ICU data received.` has something to do with not starting webview frame of the program. But I wanted to report you the last thing which the py2exe -> exe has given me when I pack it to start on the console.

To remind for others to read. My initial aim was to pack a webview used script with an exe maker. I have done this with pyinstaller with success. But then a submodule in my script was in conflict with pyinstaller and chromedriver therefore I wanted to try py2exe to pack my script. Did the packing, but not able to run to show webview component.

Maybe it would be possible, to pack and run a pywebview/webview equipped script with py2exe. But I could not do it.
I have a limited observation options to see what is happening.

Therefore the only thing I got lately is this : [0401/155724.694:ERROR:icu_util.cc(133)] Invalid file descriptor to ICU data received.

If this has nothing to do with the initiation of the webview frame or the programme total, I must give up and find other means to create a user interface maybe.

Thank you very much for all answers.
